plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -213,87 +213,100 @@ public void registerModelMeta(MLRegisterModelMetaInput mlRegisterModelMetaInput,
-                                    .algorithm(functionName)
-                                    .version(newVersion + "")
-                                    .modelGroupId(mlRegisterModelMetaInput.getModelGroupId())
-                                    .description(mlRegisterModelMetaInput.getDescription())
-                                    .modelFormat(mlRegisterModelMetaInput.getModelFormat())
-                                    .modelState(MLModelState.REGISTERING)
-                                    .modelConfig(mlRegisterModelMetaInput.getModelConfig())
-                                    .totalChunks(mlRegisterModelMetaInput.getTotalChunks())
-                                    .modelContentHash(mlRegisterModelMetaInput.getModelContentHashValue())
-                                    .modelContentSizeInBytes(mlRegisterModelMetaInput.getModelContentSizeInBytes())
-                                    .createdTime(now)
-                                    .lastUpdateTime(now)
-                                    .build();
-                                IndexRequest indexRequest = new IndexRequest(ML_MODEL_INDEX);
-                                indexRequest
-                                    .source(mlModelMeta.toXContent(XContentBuilder.builder(XContentType.JSON.xContent()), EMPTY_PARAMS));
-                                indexRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
-
-                                client.index(indexRequest, ActionListener.wrap(response -> {
-                                    log.debug("Index model meta doc successfully {}", modelName);
-                                    listener.onResponse(response.getId());
-                                }, e -> {
-                                    log.error("Failed to index model meta doc", e);
-                                    listener.onFailure(e);
-                                }));
-                            }, ex -> {
-                                log.error("Failed to init model index", ex);
-                                listener.onFailure(ex);
-                            }));
-                        }, e -> {
-                            log.error("Failed to update model group", e);
-                            listener.onFailure(e);
-                        }));
-                    } else {
-                        log.error("Model group not found");
-                        listener.onFailure(new MLResourceNotFoundException("Fail to find model group"));
-                    }
-                }, e -> {
-                    log.error("Failed to get model group", e);
-                    listener.onFailure(new MLValidationException("Failed to get model group"));
-                }));
-            } catch (Exception e) {
-                log.error("Failed to register model", e);
-                listener.onFailure(e);
+                uploadMLModelMeta(mlRegisterModelMetaInput, "1", listener);
```

Comment:
Is `1` model version? Should we do: `1.0.0`

Reply:
Also please put this as a Static variable on top of the class.

Reply:
We are actually using int for version. Maybe 1 is fine: https://github.com/opensearch-project/ml-commons/pull/1029/files/8bb508ac58194eab390e1611da22f5139c083bc9#diff-3649b3e049ca94aaaff7fa072be16a1c7f5645a04c5a2230276b46bc60cfe956R226